apps/shipments/views.py

```diff
@@ -137,35 +173,9 @@ def add_tracking_data(self, request, version, pk):
 
         return Response(status=status.HTTP_204_NO_CONTENT)
 
-    @method_decorator(cache_page(60 * 10, cache='page'))  # Cache responses for 10 minutes
-    @action(detail=True, methods=['get', 'post'], permission_classes=(IsAuthenticatedOrDevice,))
+    @action(detail=True, methods=['post'], permission_classes=(permissions.AllowAny,))
     def tracking(self, request, version, pk):
-        if request.method == 'GET':
-            return self.get_tracking_data(request, version, pk)
-        if request.method == 'POST':
-            return self.add_tracking_data(request, version, pk)
-        raise exceptions.MethodNotAllowed(request.method)
-
-    def update(self, request, *args, **kwargs):
-        """
-        Update the shipment with new details, overwriting the built-in method
-        """
-        partial = kwargs.pop('partial', False)
-        instance = self.get_object()
-        LOG.debug(f'Updating shipment {instance.id} with new details.')
-        log_metric('transmission.info', tags={'method': 'shipments.update', 'module': __name__})
-
-        serializer = ShipmentUpdateSerializer(instance, data=request.data, partial=partial,
-                                              context={'auth': get_jwt_from_request(request)})
-        serializer.is_valid(raise_exception=True)
-
-        shipment = self.perform_update(serializer)
-        async_jobs = shipment.asyncjob_set.filter(state__in=[JobState.PENDING, JobState.RUNNING])
-        response = ShipmentTxSerializer(shipment)
-
-        response.instance.async_job_id = async_jobs.latest('created_at').id if async_jobs else None
-
-        return Response(response.data, status=status.HTTP_202_ACCEPTED)
+        return self.add_tracking_data(request, version, pk)
```

Comment:
Please move the contents of add_tracking_data here, there's no need to have a separate function.